### DISCUSSION TITLE
Generating wing internal structure geometry

### DISCUSSION THREAD

**Discussion by TiagoNunes-git** [DLR-SL/CPACS#747] [Q&A] [answered]
Hello guys!

As explained in the last webinar, we are having trouble with the .xml file that we are using to export the internal structure using tigl3.

The code that we used is written in the picture below and it works to export some of the internal structure but it stops generating ribs when it gets to a zone of the front spar that apparently was not generated correctly.

![image](https://user-images.githubusercontent.com/40853750/66920709-bf8d4300-f01b-11e9-82d9-af05f6a19f48.png)

We tried using several .xml files from last exercises and the result was the same with both versions of CPACS, 2 and 3.

![image](https://user-images.githubusercontent.com/40853750/66920619-8ead0e00-f01b-11e9-88d7-c7ee45a5e4bc.png)

We think that it's probably a problem with the file that we have as an input. Either in the initial .xml file or after it was computed by OpenAD.

Can you investigate what's the problem with this file or send us another file would work?
Also, if you can include aerodynamic loads in it would e perfect so we can start working in implementing those. 

I will attach the .xml files that we used. (converted to .txt so it's possible to attach here)
[aircraft.txt](https://github.com/DLR-SL/CPACS-IFARX/files/3734380/aircraft.txt)
[D165_input.txt](https://github.com/DLR-SL/CPACS-IFARX/files/3734381/D165_input.txt)
[aircraft3.txt](https://github.com/DLR-SL/CPACS-IFARX/files/3734382/aircraft3.txt)

Thanks in advance.
Best regards,
Tiago

**Comment by MarAlder** [comment]
As you can see in the [referenced issue](https://github.com/DLR-SC/tigl/issues/643), the TiGL developers are already working on it. To continue the project we will shift the spar position `wing_Spar_FS_P2` a little bit so that you can continue with your analysis.

**Comment by TiagoNunes-git** [comment]
Thank you Marko! 
Can you please send me or place here the .xml file with that change?

**Comment by MarAlder** [comment, marked as answer]
Actually Sebastian is preparing a new input file with load cases. It might differ from the current one. However, in the current file I just interpolated linearly between `wing_Spar_FS_P2` and `wing_Spar_FS_P3` to shift the first spar position from eta=0.34 to 0.345:

[aircraft.zip](https://github.com/DLR-SL/CPACS-IFARX/files/3737986/aircraft.zip)

![fs_problem_solved](https://user-images.githubusercontent.com/43143741/66988921-a9d35880-f0c3-11e9-94b0-176d1323b240.png)

We will add a little fix of the data when we set up the workflow.

**Comment by TiagoNunes-git** [comment]
Ok. Thanks a lot for the fix. In the mean time we will work in having the internal structure added to our tool. 

Looks great! :)

![image](https://user-images.githubusercontent.com/40853750/66989841-041ceb00-f0bd-11e9-8d16-667f53524d87.png)
